text/0000-more-api-documentation-conventions.md

```diff
@@ -0,0 +1,730 @@
+- Feature Name: More API Documentation Conventions
+- Start Date: 2016-03-31
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+[RFC 505] introduced certain conventions around documenting Rust projects. This
+RFC augments that one, and a full text of the older one combined with these
+modfications is provided below.
+
+[RFC 505]: https://github.com/rust-lang/rfcs/blob/master/text/0505-api-comment-conventions.md
+
+# Motivation
+[motivation]: #motivation
+
+Documentation is an extremely important part of any project. It’s important
+that we have consistency in our documentation.
+
+For the most part, the RFC proposes guidelines that are already followed today,
+but it tries to motivate and clarify them.
+
+# Detailed design
+[design]: #detailed-design
+
+### English
+[english]: #english
+
+This section applies to `rustc` and the standard library.
+
+An additional suggestion over RFC 505: One specific rule that comes up often:
+when quoting something for emphasis, use a single quote, and put punctuation
+outside the quotes, ‘this’. When quoting something at length, “use double
+quotes and put the punctuation inside of the quote.” Most documentation will
+end up using single quotes, so if you’re not sure, just stick with them.
+
+### Using Markdown
+[using-markdown]: #using-markdown
+
+The updated list of common headings is:
+
+* Examples
+* Panics
+* Errors
+* Safety
+* Aborts
+* Undefined Behavior
+
+RFC 505 suggests that one should always use the `rust` formatting directive:
+
+    ```rust
+    println!("Hello, world!");
+    ```
+
+    ```ruby
+    puts "Hello"
+    ```
+
+But, in API documentation, feel free to rely on the default being ‘rust’:
+
+    /// For example:
+    ///
+    /// ```
+    /// let x = 5;
+    /// ```
+
+Other places do not know how to highlight this anyway, so it's not important to
+be explicit.
+
+RFC 505 suggests that references and citation should be linked ‘reference
+style.’ This is still recommended, but prefer to leave off the second `[]`:
+
+```
+[Rust website]
+
+[Rust website]: http://www.rust-lang.org
+```
+
+to
+
+```
+[Rust website][website]
+
+[website]: http://www.rust-lang.org
+```
+
+But, if the text is very long, it is okay to use this form.
+
+### Examples in API docs
+[examples-in-api-docs]: #examples-in-api-docs
+
+Everything should have examples. Here is an example of how to do examples:
```

Comment:
I can't agree on that. Examples are just way too convenient and useful to not add them. Just in the case you didn't understand the function description, you look at the example and it generally helps. Otherwise it's just convenient if you're too lazy to understand each parameter and just copy/paste.